jolt-core/src/zkvm/ram/raf_evaluation.rs

```diff
@@ -115,38 +116,98 @@ impl<F: JoltField> RafEvaluationSumcheckProver<F> {
     ) -> Self {
         let T = trace.len();
         let K = 1 << params.log_K;
-        let num_chunks = rayon::current_num_threads().next_power_of_two().min(T);
-        let chunk_size = (T / num_chunks).max(1);
-
-        // TODO(moodlezoup): reuse
-        let eq_r_cycle: Vec<F> = EqPolynomial::evals(&params.r_cycle.r);
-
-        let ra_evals: Vec<F> = trace
-            .par_chunks(chunk_size)
-            .enumerate()
-            .map(|(chunk_index, trace_chunk)| {
-                let mut result = unsafe_allocate_zero_vec(K);
-                let mut j = chunk_index * chunk_size;
-                for cycle in trace_chunk {
-                    if let Some(k) =
-                        remap_address(cycle.ram_access().address() as u64, memory_layout)
-                    {
-                        result[k as usize] += eq_r_cycle[j];
+
+        // Two-table split-eq:
+        // EqPolynomial::evals uses big-endian bit order: r_cycle[0] is MSB, r_cycle[last] is LSB.
+        // To get contiguous blocks in the cycle index, we split off the LSB half (suffix) as E_lo.
+        let r_cycle = &params.r_cycle.r;
+        let log_T = r_cycle.len();
+        let lo_bits = log_T / 2;
+        let hi_bits = log_T - lo_bits;
+        let (r_hi, r_lo) = r_cycle.split_at(hi_bits);
+
+        let (E_hi, E_lo) = rayon::join(
+            || EqPolynomial::<F>::evals(r_hi),
+            || EqPolynomial::<F>::evals(r_lo),
+        );
+
+        let in_len = E_lo.len(); // 2^lo_bits
+
+        // Split E_hi into exactly num_threads chunks to minimize allocations
+        let num_threads = rayon::current_num_threads();
+        let out_len = E_hi.len(); // 2^hi_bits
+        let chunk_size = out_len.div_ceil(num_threads);
+
+        let chunk_ranges: Vec<(usize, usize)> = (0..num_threads)
+            .map(|t| {
+                let start = t * chunk_size;
+                let end = std::cmp::min(start + chunk_size, out_len);
+                (start, end)
+            })
+            .filter(|(start, end)| start < end)
+            .collect();
```

Comment:
this seems a bit unidiomatic to me, can't we just do `E_hi.par_chunks(chunk_size).enumerate()`?